transformers/storage/cat/repository_test.go

```diff
@@ -40,112 +41,85 @@ var _ = Describe("Cat storage repository", func() {
 	})
 
 	Describe("Variable", func() {
-		var result VariableRes
+		It("panics if the metadata name is not recognized", func() {
+			unrecognizedMetadata := utils.StorageValueMetadata{Name: "unrecognized"}
+			repoCreate := func() {
+				repo.Create(diffID, fakeHeaderID, unrecognizedMetadata, "")
+			}
 
-		Describe("Live", func() {
-			It("writes a row", func() {
-				liveMetadata := utils.GetStorageValueMetadata(cat.Live, nil, utils.Uint256)
-
-				err := repo.Create(fakeHeaderID, liveMetadata, fakeUint256)
-				Expect(err).NotTo(HaveOccurred())
-
-				err = db.Get(&result, `SELECT header_id, live AS value FROM maker.cat_live`)
-				Expect(err).NotTo(HaveOccurred())
-				AssertVariable(result, fakeHeaderID, fakeUint256)
-			})
-
-			It("does not duplicate row", func() {
-				liveMetadata := utils.GetStorageValueMetadata(cat.Live, nil, utils.Uint256)
-				insertOneErr := repo.Create(fakeHeaderID, liveMetadata, fakeUint256)
-				Expect(insertOneErr).NotTo(HaveOccurred())
-
-				insertTwoErr := repo.Create(fakeHeaderID, liveMetadata, fakeUint256)
+			Expect(repoCreate).Should(Panic())
+		})
 
-				Expect(insertTwoErr).NotTo(HaveOccurred())
-				var count int
-				getCountErr := db.Get(&count, `SELECT count(*) FROM maker.cat_live`)
-				Expect(getCountErr).NotTo(HaveOccurred())
-				Expect(count).To(Equal(1))
-			})
+		Describe("Live", func() {
+			liveMetadata := utils.GetStorageValueMetadata(cat.Live, nil, utils.Uint256)
+			inputs := shared_behaviors.StorageBehaviorInputs{
+				ValueFieldName:   cat.Live,
+				Value:            fakeUint256,
+				StorageTableName: "maker.cat_live",
```

Comment:
Definitely not a part of this PR but writing it down so I don't forget, seems like we could restructure this struct to accept a `Schema` and `TableName` and then use the constants from `transformers/shared/constants/table.go`

Reply:
👍 definitely like that idea! I'll plan to do this in another PR.